src/main/java/io/debezium/connector/spanner/db/model/schema/SpannerSchema.java

```diff
@@ -51,9 +51,19 @@ public void addColumn(String tableName, String columnName, String type, long ord
                 columns = new ArrayList<>();
                 tableMap.put(tableName, columns);
             }
-            if (!columns.stream().anyMatch(column -> column.getName().equals(columnName))) {
-                columns.add(Column.create(columnName, type, primaryKey, ordinalPosition, nullable));
+            columns.add(Column.create(columnName, type, tableToPrimaryColumnMap.get(tableName).contains(columnName), ordinalPosition, nullable));
```

Comment:
How do we ensure against adding a primary column both in addColumn and addPrimaryColumn?

Reply:
addColumn contains all columns **including primary column**, to figure out whether a column is a primary column or not only addPrimaryColumn is used. (this is like application level join)

Example :- 
tableToPrimaryColumnMap - [["table1"->["pk1", "pk2"]]]
tableMap - [["table1"->["pk1", "pk2", "col1", "col2"]]]

while creating column **tableToPrimaryColumnMap.get("table1").contains("pk1")** is used to figure out whether a column is primary or not. 

So **tableToPrimaryColumnMap** is just to keep info about primary columns and only used when **Column.create()** is called.

Reply:
Looks great, thanks for the change.